lib/msf/ui/console/module_option_tab_completion.rb

```diff
@@ -338,19 +335,30 @@ def option_values_target_addrs(mod)
         # Provide the target ports
         #
         def option_values_target_ports(mod)
-          res = [ ]
-          return res if !framework.db.active
-          return res if !mod.datastore['RHOST']
+          res = []
+          return res unless framework.db.active
+          return res if mod.datastore['RHOST'].nil?
 
-          host = framework.db.has_host?(framework.db.workspace, mod.datastore['RHOST'])
-          return res if !host
+          host_addresses = []
+          mod.datastore['RHOST'].split.each { |addr|
+            host_addresses << addr.split('%', 2)
+          }
 
-          framework.db.services.each do |service|
-            if service.host_id == host.id
+          hosts = []
+          host_addresses.each do |addr|
+            host = framework.db.get_host({:address => addr, :workspace => framework.db.workspace})
```

Comment:
I wonder if this can use `hosts` directly? It appears to support filtering by address

https://github.com/rapid7/metasploit-framework/blob/0f433cf2ef739db5f7865ba4d5d36f301278873b/lib/msf/core/db_manager/host.rb#L139-L151